snuba/query/snql/__init__.py

```diff
@@ -1,28 +1,255 @@
+        return Query({}, None, collect, None, None, None, groupby, None, orderby)
+
+    def visit_function_name(self, node: Node, visited_children: Iterable[Any]) -> str:
+        return visit_function_name(node, visited_children)
+
+    def visit_column_name(self, node: Node, visited_children: Iterable[Any]) -> Column:
+        return visit_column_name(node, visited_children)
+
+    def visit_low_pri_tuple(
+        self, node: Node, visited_children: Tuple[LowPriOperator, Expression]
+    ) -> LowPriTuple:
+        return visit_low_pri_tuple(node, visited_children)
+
+    def visit_high_pri_tuple(
+        self, node: Node, visited_children: Tuple[HighPriOperator, Expression]
+    ) -> HighPriTuple:
+        return visit_high_pri_tuple(node, visited_children)
+
+    def visit_low_pri_op(
+        self, node: Node, visited_children: Iterable[Any]
+    ) -> LowPriOperator:
+        return visit_low_pri_op(node, visited_children)
+
+    def visit_high_pri_op(
+        self, node: Node, visited_children: Iterable[Any]
+    ) -> HighPriOperator:
+        return visit_high_pri_op(node, visited_children)
+
+    def visit_arithmetic_term(
+        self, node: Node, visited_children: Tuple[Any, Expression, Any]
+    ) -> Expression:
+        return visit_arithmetic_term(node, visited_children)
+
+    def visit_low_pri_arithmetic(
+        self,
+        node: Node,
+        visited_children: Tuple[Any, Expression, Any, LowPriArithmetic],
+    ) -> Expression:
+        return visit_low_pri_arithmetic(node, visited_children)
+
+    def visit_high_pri_arithmetic(
+        self,
+        node: Node,
+        visited_children: Tuple[Any, Expression, Any, HighPriArithmetic],
+    ) -> Expression:
+        return visit_high_pri_arithmetic(node, visited_children)
+
+    def visit_numeric_literal(
+        self, node: Node, visited_children: Iterable[Any]
+    ) -> Literal:
+        return visit_numeric_literal(node, visited_children)
+
+    def visit_quoted_literal(
+        self, node: Node, visited_children: Tuple[Any, Node, Any]
+    ) -> Literal:
+        return visit_quoted_literal(node, visited_children)
+
+    def visit_order_by_clause(self, node, visited_children):
+        _, _, order_columns, _ = visited_children
+        return order_columns
+
+    # order_columns* (low_pri_arithmetic (("ASC"/"DESC"))
+    def visit_order_list(self, node, visited_children):
+        left_order_list, right_order, order = visited_children
+
+        ret: List[OrderBy] = []
+        if not isinstance(left_order_list, Node):
+            if not isinstance(left_order_list, (list, tuple)):
+                ret.append(left_order_list)
+            else:
+                for p in left_order_list:
+                    ret.append(p)
+
+        if order.text == "ASC":
+            ret.append(OrderBy(OrderByDirection.ASC, right_order))
+        elif order.text == "DESC":
+            ret.append(OrderBy(OrderByDirection.DESC, right_order))
+
+        return ret
```

Comment:
Why would you ever get to this statement ?

Reply:
Grammar has the structure of:

order_list            = order_columns* low_pri_arithmetic ("ASC"/"DESC")

so the righthand side (low_pri_arithmetic ("ASC"/"DESC") is not being processed as order_columns. In order to add it to the order_by list, then, we need to process it in visit_order_list and take a look at whether it is ASC/DESC.